File: Dispatcher/MessageApiConverter.m
//
//  MessageApiConverter.m
//  POS
//
//  Created by Gal Blank on 11/30/15.
//  Copyright © 2015 1stPayGateway. All rights reserved.
//

#import "MessageApiConverter.h"



@implementation MessageApiConverter

static  MessageApiConverter *sharedApiConverterInstance = nil;

+ (id)sharedInstance
{
    static dispatch_once_t once;
    static id sharedInstance;
    dispatch_once(&once, ^{
        sharedInstance = [[self alloc] init];
    });
    return sharedInstance;
}


+ (id)allocWithZone:(NSZone *)zone {
    @synchronized(self) {
        if (sharedApiConverterInstance == nil) {
            sharedApiConverterInstance = [super allocWithZone:zone];
            // assignment and return on first allocation
            return sharedApiConverterInstance;
        }
    }
    // on subsequent allocation attempts return nil
    return nil;
}

- (id)copyWithZone:(NSZone *)zone
{
    return self;
}

-(id)init
{
    if (self = [super init]) {
    
    }
    return self;
}




@end

File: Dispatcher/MessageApiConverter.swift
//
//  MessageApiConverter.m
//  POS
//
//  Created by Gal Blank on 11/30/15.
//  Copyright © 2015 1stPayGateway. All rights reserved.
//

import Foundation

class MessageApiConverter:NSObject  {
    
    static let sharedInstance = MessageApiConverter()
    
    func messageTypeToApiCall(_ msg:Message) {
        
        switch (msg.messageFromRoutingKey()) {
        case "messageTypeTokenForTransaction":
            msg.messageApiEndPoint = "Transaction/GenerateTokenForTransaction"
            msg.httpMethod = "get";
            break
        case "messageTypeOnHoldOrdersBatchPost":
            
            msg.messageApiEndPoint = ""
            msg.httpMethod = "batchPost";
            break
        case "messageTypeOnHoldOrdersSinglePost":
            
            msg.messageApiEndPoint = ""
            msg.httpMethod = "post";
            break
        case "messageTypeOnHoldOrdersRetrieveCompacted":
            
            msg.messageApiEndPoint = ""
            msg.httpMethod = "get";
            break
        default:
            
            msg.messageApiEndPoint = ""
            msg.httpMethod = "post";
            break
        }
        msg.routingKey = "api.*"
    }
}
